# Identification and characterization of *Faecalibacterium* prophages rich in diversity-generating retroelements


## Folders

* __UHGG_database__ - download UHGG genomes, identify CRISPR spacers

* __human_genome_ref, Viral_RefSeq_220, IMG_VR_2022-12-19_7.1, Pfam_36.0__ - download databases

* __Fprau_raw_reads, Fprau_clean_reads__ - clean *Faecalibacterium* strains reads

* __Fprau_assemblies__ - assemble *Faecalibacterium* strains genomes, assign taxonomy

* __VLP_reads__ - download and clean reads from human gut viral metagenomes

* __Fprau_map_reads__ - map reads to *Faecalibacterium* contigs

* __Fprau_proteomes__ - predict ORFs in *Faecalibacterium* contigs

* __Fprau_prophage_tools__ - screen *Faecalibacterium* contigs for prophages using Cenote-Taker2, VirSorter2 and geNomad

* __Fprau_contig_maps__ - build *Faecalibacterium* contig maps that help identify prophages

* __CPs_refine_coo__ - refine candidate prophage (CP) coordinates based on similarity with complete phage genomes from databases

* __CPs_clusters__ - cluster CPs into species-level clusters

* __CPs_proteomes__ - predict and annotate proteomes of CPs

* __Induction_clean_reads, Induction_map_reads, Induction_assembly__ - clean, map to bacterial genomes, and assemble reads following prophage induction experiments

* __Mushu_transposition_sites__ - analyze transposable phage Mushu termini and transposition sites based on BWA-MEM split reads

* __Phages_clusters__ - cluster induced phages and CPs with known phages using vConTACT2

* __Phages_proteomes__ - predict and annotate proteomes of the phages

* __Phages_tRNA__ - predict tRNA genes of the phages

* __Phages_genome_maps__ - plot genome maps of the phages

* __Phages_map_reads__ - map reads, obtained following the prophage induction experiments, to the genomes of the induced phages

* __Phages_hosts__ - predict hosts of the phages

* __Phages_cohorts__ - estimate prevalence of the phages in human cohorts

* __DGR_detection__ - detect DGR repeats in phage genomes

* __DGR_activity__ - access DGR activity based on MSA and reads variation

* __DGR_targets__ - annotate DGR target proteins
